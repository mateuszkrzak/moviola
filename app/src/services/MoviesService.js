import Axios from 'axios';

export const API_URL = 'http://react-cdp-api.herokuapp.com/movies';

async function getMovies(options) {
  const response = await Axios.get(API_URL, {
    params: {
      ...options,
      sortOrder: 'desc',
    },
  });

  return response;
}

async function getMoviesByGenres(genres) {
  const params = new URLSearchParams();
  genres.forEach(value => params.append('filter', value));

  const response = await Axios.get(API_URL, { params });

  return response;
}

async function getMovieById(id) {
  const response = await Axios.get(`${API_URL}/${id}`);

  return response;
}

export default { getMovies, getMoviesByGenres, getMovieById };
