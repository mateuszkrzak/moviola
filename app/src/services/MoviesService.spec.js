import Axios from 'axios';
import MoviesService, { API_URL } from './MoviesService';

jest.mock('axios');

describe('Movies service', () => {
  it('getMovies should call axios with passed options', async () => {
    Axios.get.mockImplementationOnce(() => ({}));

    MoviesService.getMovies({
      testOption: 'testValue',
      anotherTestOption: 'anotherTestValue',
    });

    expect(Axios.get).toMatchSnapshot();
  });

  it('getMovieById should call axios with passed options', async () => {
    Axios.get.mockImplementationOnce(() => ({}));

    MoviesService.getMovieById(1);
    expect(Axios.get).toMatchSnapshot();
  });

  it('getMoviesByGenres should call axios with passed options', async () => {
    Axios.get.mockImplementationOnce(() => ({}));

    MoviesService.getMoviesByGenres(['Test genre', 'Another test genre']);
    expect(Axios.get).toMatchSnapshot();
  });
});
