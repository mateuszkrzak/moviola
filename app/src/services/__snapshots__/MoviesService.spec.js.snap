// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies service getMovieById should call axios with passed options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "http://react-cdp-api.herokuapp.com/movies",
      Object {
        "params": Object {
          "anotherTestOption": "anotherTestValue",
          "sortOrder": "desc",
          "testOption": "testValue",
        },
      },
    ],
    Array [
      "http://react-cdp-api.herokuapp.com/movies/1",
    ],
  ],
}
`;

exports[`Movies service getMovies should call axios with passed options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "http://react-cdp-api.herokuapp.com/movies",
      Object {
        "params": Object {
          "anotherTestOption": "anotherTestValue",
          "sortOrder": "desc",
          "testOption": "testValue",
        },
      },
    ],
  ],
}
`;

exports[`Movies service getMoviesByGenres should call axios with passed options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "http://react-cdp-api.herokuapp.com/movies",
      Object {
        "params": Object {
          "anotherTestOption": "anotherTestValue",
          "sortOrder": "desc",
          "testOption": "testValue",
        },
      },
    ],
    Array [
      "http://react-cdp-api.herokuapp.com/movies/1",
    ],
    Array [
      "http://react-cdp-api.herokuapp.com/movies",
      Object {
        "params": URLSearchParams {
          Symbol(impl): URLSearchParamsImpl {
            "_list": Array [
              Array [
                "filter",
                "Test genre",
              ],
              Array [
                "filter",
                "Another test genre",
              ],
            ],
            "_url": null,
            Symbol(wrapper): [Circular],
          },
        },
      },
    ],
  ],
}
`;
